File: Ruby/Daily003_easy/Caesar.rb
## CAESAR CIPHER
##
## challenge #3 (easy)
## http://www.reddit.com/r/dailyprogrammer/comments/pkw2m/2112012_challenge_3_easy/
## 
## 
## sarthak7u@gmail.com
##

# returns 0 if lowercase and 1 if uppercase
# returns -1 if not alpha
def casecheck(char)

	if (char.ord >= 65 and char.ord <= 90)
		return 1
	elsif (char.ord >= 97 and char.ord <= 122)
		return 0
	else
		return -1
	end

end
		
def encrypt(text,key)

	cipherText = ""
	(text.split("")).each do |i|

		# adds character to cipher text if not num
		if casecheck(i) == -1
			cipherText += i
		# ciphers and then adds to ciphertext
		elsif casecheck(i) == 1
			cipherText += ((i.ord - 'A'.ord + key)%26 + 'A'.ord).chr
		elsif casecheck(i) == 0
			cipherText += ((i.ord - 'a'.ord + key)%26 + 'a'.ord).chr
		end

	end
	return cipherText
end

print "Enter a string to encrypt : "
text = $stdin.gets.chomp
print "Enter the key for encryption : "
key = $stdin.gets.chomp.to_i
puts encrypt(text,key)

File: Python/Daily003_easy/Caesar.py
## CAESAR CIPHER
##
## challenge #3 (easy)
## http://www.reddit.com/r/dailyprogrammer/comments/pkw2m/2112012_challenge_3_easy/
##
##
## sarthak7u@gmail.com
##

# encrypts
def encrypt(text, key):

    cipherText = ""
    i = 0

    for letter in text:
        #skips over non-alpha characters
        if letter.isalpha() == False:
            cipherText += letter
        #ciphers using caesar cipher
        else:
            if letter.isupper() == True:
                tempLetter = chr((ord(letter) - ord("A") + key)%26 + ord("A"))
            elif letter.islower() == True:
                tempLetter = chr((ord(letter) - ord("a") + key)%26 + ord("a"))
            cipherText += tempLetter
        i += 1
    return cipherText

# decrypts
def decrypt(cipherText, key):
    text = ""

    i = 0
    for letter in cipherText:
        #skips over non-alpha characters
        if letter.isalpha() == False:
            text += letter
        #ciphers using caesar cipher
        else:
            if letter.isupper() == True:
                if ord(letter) - ord('A') - key < 0:
                    tempLetter = ord(letter) - key + 26
                else:
                    tempLetter = ord(letter) - key
            elif letter.islower() == True:
                if ord(letter) - ord('a') - key < 0:
                    tempLetter = ord(letter) - key + 26
                else:
                    tempLetter = ord(letter) - key
            text += chr(tempLetter)
        i += 1
    return text


# asks user whether to decrypt or encrypt then does so
choice = str(raw_input("Enter 1 for encryption, 2 for decryption: "))
if choice == "2":
    inputText = str(raw_input("Enter some text for me to Decrypt: "))
    key = int(raw_input("Enter a key to Decrypt: "))
    print decrypt(inputText, key)
else:
    inputText = str(raw_input("Enter some text for me to Encrypt: "))
    key = int(raw_input("Enter a key to Encrypt: "))
    print encrypt(inputText, key)
